File: bibliotheque-frontend/src/app/_model/borrow.ts
// Les dates arrivent du backend sous forme de chaînes dd-MM-yyyy
// (JsonDataSerializer).
export class Borrow {
    borrowId: number;
    bookId: number;
    userId: number;
    issueDate: Date | string;
    returnDate: Date | string | null;
    dueDate: Date | string;
}

File: bibliotheque-backend/src/main/java/com/ibizabroker/bibliotheque/emprunts/internal/Borrow.java
package com.ibizabroker.bibliotheque.emprunts.internal;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

/**
 * Emprunt persisté. Jamais reçu ni renvoyé tel quel par l'API : les
 * contrôleurs passent par BorrowRequest / ReturnBorrowRequest / BorrowResponse.
 */
@Data
@Entity
@Table(name = "borrow")
public class Borrow {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer borrowId;

    Integer bookId;

    Integer userId;

    LocalDateTime issueDate;

    LocalDateTime returnDate;

    LocalDateTime dueDate;

}
